pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -471,6 +481,14 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		}
 	}
 
+	availabilitySetName := s.scope.Machine.Labels[MachineSetLabelName]
+	// Delete the availability set with the given name if no virtual machines are attached to it.
+	if err := s.availabilitySetsSvc.Delete(ctx, availabilitysets.Spec{
+		Name: availabilitySetName,
```

Comment:
Have feeling that this name will not match with AS name which was created.

https://github.com/openshift/cluster-api-provider-azure/pull/225/files#diff-af805e1e45f03df0b5b56ff4413e5ad52cd31904a94d37e8e916751953e4687dR730